test/e2e/apimachinery/watchlist.go

```diff
@@ -85,8 +89,57 @@ var _ = SIGDescribe("API Streaming (aka. WatchList)", framework.WithSerial(), fe
 		framework.ExpectNoError(err)
 		verifyStore(ctx, f, secretInformer.GetStore())
 	})
+	ginkgo.It("should be requested by client-go's List method when WatchListClient is enabled", func(ctx context.Context) {
+		featuregatetesting.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), utilfeature.DefaultFeatureGate, featuregate.Feature(clientfeatures.WatchListClient), true)
+
+		ginkgo.By(fmt.Sprintf("Adding 5 secrets to %s namespace", f.Namespace.Name))
+		var expectedSecrets []v1.Secret
+		for i := 1; i <= 5; i++ {
+			secret, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Create(ctx, newSecret(fmt.Sprintf("secret-%d", i)), metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+			expectedSecrets = append(expectedSecrets, *secret)
+		}
+
+		var actualRequestsMadeByKubeClient []string
+		clientConfig := f.ClientConfig()
+		clientConfig.Wrap(func(rt http.RoundTripper) http.RoundTripper {
+			return roundTripFunc(func(req *http.Request) (*http.Response, error) {
+				actualRequestsMadeByKubeClient = append(actualRequestsMadeByKubeClient, req.URL.RawQuery)
+				return rt.RoundTrip(req)
+			})
+		})
+		wrappedKubeClient, err := kubernetes.NewForConfig(clientConfig)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Streaming secrets from the server")
+		secretList, err := wrappedKubeClient.CoreV1().Secrets(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Verifying if the secret list was properly streamed")
+		streamedSecrets := secretList.Items
+		sort.Sort(byName(expectedSecrets))
+		sort.Sort(byName(streamedSecrets))
```

Comment:
The streamed ones should already be sorted (it's the semantic of List request).
Let's not sort them.

Reply:
good point, will update.